### PR TITLE
refactor!: rename "playground" => "dev"

### PR DESCRIPTION
**Pull request by justinmk**
Problem:
"playground" is new jargon that overlaps with existing concepts: "view" ~~"scratch"~~ and "dev" (`:help dev`).

Solution:
We should consistently use "dev" as the namespace for where "developer tools" live (it's more generalizable to other modules, whereas "playground" is more narrow). For purposes of a "throwaway sandbox object", we can use "view" ~~"scratch"~~.
- Rename `TSPlayground` => `TSView`
- Rename `playground.lua` => `dev.lua`

## Note

Didn't do a `vim.deprecate()` cycle since this is not a critical module.

### CODE REVIEW

**Comment by clason** [comment]
> TSPlayground

That's not a thing; we only have `:Inspect` and `InspectTree`; `TSPlayground` was never exposed or documented.

I'd prefer _not_ to use `TSScratch` for this, since the tree is not editable -- I would reserve it for the live query editor (which is the last missing ingredient from the treesitter playground). If you are on a crusade against playgrounds, better to name it `TSTreeView`.

No opinion about renaming the module, since that is not user-facing (we have the commands for that).

**Comment by justinmk** [comment]
> If you are on a crusade against playgrounds, better to name it `TSTreeView`.

Not against anything. But strongly in favor of encouraging consistency and avoiding entropy where we can. If we agree on that, let's frame it as something easier than "crusade", "battle", etc. 

"View" is a good idea, I'll use that instead of "scratch".

File: runtime/doc/treesitter.txt [hunk 1]
```diff
@@ -1218,9 +1218,9 @@ LanguageTree:trees({self})                              *LanguageTree:trees()*
 
 
 ==============================================================================
-Lua module: vim.treesitter.playground              *lua-treesitter-playground*
```
Comment (clason) [inline runtime/doc/treesitter.txt:1221]: We should not include this module at all in our documentation at the moment (if at all); the only function `vim.treesitter.dev.inspect_tree` is exported and more fully documented as `vim.treesitter.inspect_tree`.

File: runtime/lua/vim/treesitter/dev.lua [hunk 2]
```diff
@@ -1,16 +1,16 @@
 local api = vim.api
 
----@class TSPlaygroundModule
+---@class TSDevModule
 local M = {}
 
----@class TSPlayground
+---@class TSView
```
Comment (clason) [inline runtime/lua/vim/treesitter/dev.lua:6]:
```suggestion
---@class TSTreeView
```
This is specifically a view of the LanguageTree, not a generic view (e.g. of queries).

**Comment by justinmk** [comment]
@clason I feel naked without your 👍 🥇 🚀

**Comment by justinmk** [comment]
asan failure, anyone seen this before?

```
FAILED   test/functional/plugin/health_spec.lua @ 33: :checkhealth completions can be listed via getcompletion()
test/functional/plugin/health_spec.lua:36: Expected objects to be the same.
Passed in:
(nil)
Expected:
(string) 'provider'

ERROR    2 errors, listed below:
ERROR    test/functional/terminal/tui_spec.lua @ 78: TUI rapid resize #7572 #7628
test/functional/ui/screen.lua:178: EOF was received from Nvim. Likely the Nvim process crashed.

stack traceback:
	test/functional/ui/screen.lua:178: in function 'try_resize'
	test/functional/ui/screen.lua:231: in function 'try_resize'
	test/functional/terminal/tui_spec.lua:95: in function <test/functional/terminal/tui_spec.lua:78>

...

-- Output to stderr:
1 lua references were leaked!1 lua references were leaked!
```

**Comment by zeertzjq** [comment]
#23762

**Review by clason — APPROVED** [review]
LGTM 👍 🚀 🏅

**Pull request merged**